### DISCUSSION TITLE
Rewrite UI webdriver-ts-results to Radix UI | Shadcn UI, TanStack Table and D3

### DISCUSSION THREAD

**Discussion by nakrovati** [krausest/js-framework-benchmark#1366] [Ideas]
Why?
- To improve UI
- Simplify table logic
- Simplify further development
- Reduce bundle size

It's not some mandatory action, but it would be nice to have the site display normally on phones, have a dark theme and not load megabytes of plotly code.

**Comment by krausest** [comment]
If you want to do a complete rewrite I suggest adding it as a second implementation for the results and keep the old one at least for a while. 
What I like(d) about the old implementation is that the maintenance over the years was pretty much ok.

**Reply by leeoniya** [reply]
i would be on board with re-implementing the results table in `ivi`, because ivi is _awesome_.

don't be fooled by its keyed implementation here, the templates can be html strings: https://github.com/localvoid/ivi/blob/master/packages/@ivi/htm/README.md

**Reply by nakrovati** [reply]
Anyway, here's an even better idea. Simply rewrite everything on Astro + Svelte. If we are going to rewrite ui, we should do it in a big way.

**Reply by nakrovati** [reply]
But first you need to decide what to do with the transition to vite.

**Reply by leeoniya** [reply]
eh, svelte is gonna be bigger and slower than ivi. ivi does have astro support, but i dont see a need to make things more complex than they need to be. the results ui is a pretty simple thing.

**Reply by krausest** [reply]
If it only were vite a PR might be worth it, but I'm not sure about the further changes. For those I clearly prefer a second implementation since I'm not sure (yet) about the long term maintenance.

**Reply by nakrovati** [reply]
Well astro I wanted to take to glue all the site routes: `/`, `/current/` and `/{year}/{benchmarkName}`. And Svelte I just know, besides it weighs much less than React, that's why I decided to take it.

**Reply by nakrovati** [reply]
@krausest  Well, then I can drop the PR Vite now. 

But as I wrote in the issue, there are two problems:

1. Either Vite with vite-singlefile-html plugin generates 1 big html on 2mb (react+zustand+jstat+plotly+results data)
2. Or it divides them into chunks by default. So you can't view the resulting index.html in a browser without starting the server (because of CORS). 

P.S: I've added the ability to create a large html or regular build in different scripts if needed.

P.S 2: That's actually why I wanted to take the Astro.

**Reply by krausest** [reply]
I prefer option 2. I switched my workflow (just yesterday) such that the results could be loaded via the server again.
Are the chunks stable from build to build?

**Reply by nakrovati** [reply]
Yes, I manually chunked everything to remove the hashes in the file names. Although everything works with these hashes too.

**Reply by leeoniya** [reply]
why chunk anything, honestly?

when Plotly and React are removed, the single-file bundle will be probably ~70kb, with the json data comprising the remaining 666kb [1], either fetched async or just bundled into the single file. (that results file can probably be reduced by 30-50%, too). then, after gzip you're looking at probably 200kb for the entire thing. :shrug: 

people are too obsessed with the latest in frontend tooling hype, usually for no good reason.

[1] https://github.com/krausest/js-framework-benchmark/blob/bd4a0c9d5ef4065232b25cfd0092f4308ca72ed5/webdriver-ts/results.json

**Reply by nakrovati** [reply]
@leeoniya That's what I wanted to do, just when migrating to vite krausets said to split the code into chunks like now (`index.js` + `BoxPlotTable.js` + `plotly.js`). 

You're suggesting to store the results in a folder and use react-router to load them when the route changes, right? I suggested to do the same with Astro, just adding processing of routes `/` and `/current/` to webdriver-ts-results repo.

**Reply by leeoniya** [reply]
if the total app size with results < 200kb, then just inline the whole thing into a single html file. no router, no fetch, no server, no nothing.

**Comment by krausest** [comment]
Last year I played a bit with angular, lit, preact, solid and svelte and created a version with the result app. Then I found it funny to make a benchmark driver to test some operations on that table :)
(But I never made a nice looking results table from those results..)
I thought I keep that repo as a joke, but maybe I'll try to run it with the new zustand version or that new implementation.

**Comment by nakrovati** [comment]
Nice joke, as much as 6k stars.
